src/commands.rs

```diff
@@ -3950,53 +3955,81 @@ fn is_fast_forward(repo: RepoRef, branch_name: &str, new_target_id: &CommitId) -
 
 fn cmd_branch(ui: &mut Ui, command: &CommandHelper, args: &BranchArgs) -> Result<(), CommandError> {
     let mut workspace_command = command.workspace_helper(ui)?;
-    let branch_name = &args.name;
+    let branch_names: Vec<&str> = if args.delete || args.forget {
+        args.names
+            .iter()
+            .map(|branch_name| {
+                match workspace_command
+                    .repo()
+                    .view()
+                    .get_local_branch(branch_name)
+                {
+                    Some(_) => Ok(branch_name.as_str()),
+                    None => Err(CommandError::UserError(format!(
+                        "No such branch: {}",
+                        branch_name
+                    ))),
+                }
+            })
+            .try_collect()?
+    } else {
+        args.names.iter().map(|name| name.as_str()).collect()
+    };
+
+    if branch_names.is_empty() {
+        ui.write_warn("warn: no branches provided\n")?;
```

Comment:
Actually, can this be achieved by annotating the `BranchArgs::names` field with `#[clap(required = true)]`?

Reply:
This was actually intentional as written. The idea is to support automated workflows, like if you have something like this in a script:

```
set -e
branches=$(jj branches | grep 'foo' | ...)
jj branch --delete $branches
```

So if it happens not to delete any branches, it doesn't cause the script to fail. On the other hand, a warning is helpful in the interactive case like if you write

```
$ jj branch --delete $(jj branches | grep 'foo' | ...)
```

and it didn't actually return any results.

Reply:
Ah, good point!